src/pages/homepage/HomeProductCard.jsx

```diff
@@ -0,0 +1,18 @@
+import { Link } from 'react-router-dom'
+import { useProductFilter } from '../../context/index.jsx'
```

Comment:
You can use absolute paths in imports, it might sound hard to implement but its easy and will reduce your work a lot, making your code look better at the same time.

https://stackoverflow.com/questions/41993461/import-react-components-with-absolute-path
you can refer to this if you wish to implement.

Reply:
I have tried doing this but was not working so leaving this, for now, will do it later on.